IdentityCore/src/oauth2/aad_v2/MSIDAADV2Oauth2Factory.m

```diff
@@ -98,14 +98,27 @@ - (BOOL)verifyResponse:(MSIDAADV2TokenResponse *)response
     {
         if (response.error && error)
         {
+            MSIDErrorCode errorCode = response.oauthErrorCode;
+            NSDictionary *additionalUserInfo = nil;
+            
+            /* This is a special error case for True MAM,
+             where a combination of unauthorized client and MSID_PROTECTION_POLICY_REQUIRED should produce a different error */
+            MSIDErrorCode oauthErrorCode = MSIDErrorCodeForOAuthError(response.error, MSIDErrorServerOauth);
+            if (oauthErrorCode == MSIDErrorServerUnauthorizedClient
+                && [response.suberror isEqualToString:MSID_PROTECTION_POLICY_REQUIRED])
+            {
+                errorCode = MSIDErrorServerProtectionPoliciesRequired;
+                additionalUserInfo = @{MSIDUserDisplayableIdkey : response.additionalUserId ?: @""};
```

Comment:
Can this be moved to base AAD factory, as it should be applicable to both v1 and v2?